gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/ConfigurableCleanableDataset.java

```diff
@@ -67,11 +67,25 @@ public ConfigurableCleanableDataset(FileSystem fs, Properties jobProps, Path dat
     Preconditions.checkArgument(config.hasPath(RETENTION_POLICY_CLASS_KEY), "Missing property " + RETENTION_POLICY_CLASS_KEY);
 
     try {
-      this.versionFinder =
-          (VersionFinder<? extends T>) ConstructorUtils.invokeConstructor(Class.forName(config.getString(VERSION_FINDER_CLASS_KEY)),
-              this.fs, config);
-      this.retentionPolicy =
-          (RetentionPolicy<T>) ConstructorUtils.invokeConstructor(Class.forName(config.getString(RETENTION_POLICY_CLASS_KEY)), config);
+
+      if (ConstructorUtils.getAccessibleConstructor(Class.forName(config.getString(VERSION_FINDER_CLASS_KEY)),
```

Comment:
Can we make a utility method to do this? E.g. `GobblinConstructorUtils.invokeFirstConstructor(Class<?>, List<Object>... options)`